### PR TITLE
refactor to composite action

### PR DESCRIPTION
**Pull request by renefritze**

### CODE REVIEW

**Comment by platisd** [comment]
Good work. Do you have a link to a successful run?

Also, from what I understand, the users don't need to change anything when they use the Action. It will run in whatever environment they have launched, correct?

**Comment by renefritze** [comment]
> Good work. Do you have a link to a successful run?
>
Thanks. I've only tested it on a private project. I'll have a think on how this could be generally tests in CI here.

> Also, from what I understand, the users don't need to change anything when they use the Action. It will run in whatever environment they have launched, correct?

Generally, yeah. If they are already setting up python, they'd need to pass in their version, else the setup-action will clash.

File: action_launcher.bash [hunk 1]
```diff
@@ -27,7 +27,7 @@ if [ ! -f "$INPUT_CLANG_TIDY_FIXES" ]; then
   exit 0
 fi
 
-/action/run_action.py \
+${GITHUB_ACTION_PATH}/run_action.py \
```
Comment (oleg-derevenetz) [inline action_launcher.bash:30]: Hi @renefritze what if `GITHUB_ACTION_PATH` contains a space or a special character like `?` or `*`?

**Comment by platisd** [comment]
> Generally, yeah. If they are already setting up python, they'd need to pass in their version, else the setup-action will clash.

@renefritze :thinking: can we do anything to avoid a new major version, since this is a breaking change?

**Comment by platisd** [comment]
> > Generally, yeah. If they are already setting up python, they'd need to pass in their version, else the setup-action will clash.
> 
> @renefritze 🤔 can we do anything to avoid a new major version, since this is a breaking change?

Some follow-up questions I just thought:
- What are the consequences of such "clash"? That after running this action the user will end up with a potentially different Python version than the one they had set up?
- If there's no way to avoid this clash, what's the best way to make it very clear to the users this may happen and therefore they need to explicitly set the python version?
  - Potentially make the Python version of _this_ action a **required** argument?

**Comment by renefritze** [comment]
> Some follow-up questions I just thought:
> 
> * What are the consequences of such "clash"? That after running this action the user will end up with a potentially different Python version than the one they had set up?

I don't know. From the [docs](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#specifying-multiple-pythonpypy-version) on setting multiple versions, I would think the last version "wins", yeah.

>   * Potentially make the Python version of _this_ action a **required** argument?

That would work, yes. 
I might just have had a better idea that isolates this action's setup entirely: https://github.com/platisd/clang-tidy-pr-comments/pull/62/commits/51a21cbfd881b87371c751ec014c92cf2866fbb6

**Comment by platisd** [comment]
> > Some follow-up questions I just thought:
> > 
> > * What are the consequences of such "clash"? That after running this action the user will end up with a potentially different Python version than the one they had set up?
> 
> I don't know. From the [docs](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#specifying-multiple-pythonpypy-version) on setting multiple versions, I would think the last version "wins", yeah.
> 
> > * Potentially make the Python version of _this_ action a **required** argument?
> 
> That would work, yes. I might just have had a better idea that isolates this action's setup entirely: [51a21cb](https://github.com/platisd/clang-tidy-pr-comments/commit/51a21cbfd881b87371c751ec014c92cf2866fbb6)

```
${GITHUB_ACTION_PATH}/venv/python -m pip install -r ${GITHUB_ACTION_PATH}/requirements.txt
```

Did this work?

File: action_launcher.bash (hunk 1 above)
Comment (platisd) [reply action_launcher.bash]: Has this been addressed @renefritze btw?

Comment (renefritze) [reply action_launcher.bash]: Quoted the paths, yes.

**Comment by renefritze** [comment]
> > > Some follow-up questions I just thought:
> > > 
> > > * What are the consequences of such "clash"? That after running this action the user will end up with a potentially different Python version than the one they had set up?
> > 
> > 
> > I don't know. From the [docs](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#specifying-multiple-pythonpypy-version) on setting multiple versions, I would think the last version "wins", yeah.
> > > * Potentially make the Python version of _this_ action a **required** argument?
> > 
> > 
> > That would work, yes. I might just have had a better idea that isolates this action's setup entirely: [51a21cb](https://github.com/platisd/clang-tidy-pr-comments/commit/51a21cbfd881b87371c751ec014c92cf2866fbb6)
> 
> ```
> ${GITHUB_ACTION_PATH}/venv/python -m pip install -r ${GITHUB_ACTION_PATH}/requirements.txt
> ```
> 
> Did this work?

Not right away :)
https://github.com/platisd/clang-tidy-pr-comments/actions/runs/7531944101/job/20501597277?pr=62

File: action.yml [hunk 2]
```diff
@@ -24,9 +24,33 @@ inputs:
...
+    - name: Run action
+      run: ${GITHUB_ACTION_PATH}/action_launcher.bash
+      shell: bash -l {0}
+      env:
+        INPUT_GITHUB_TOKEN: ${{ inputs.github_token }}
+        INPUT_CLANG_TIDY_FIXES: ${{ inputs.clang_tidy_fixes }}
+        INPUT_PULL_REQUEST_ID: ${{ inputs.pull_request_id }}
+        INPUT_REQUEST_CHANGES: ${{ inputs.request_changes }}
+        INPUT_SUGGESTIONS_PER_COMMENT: ${{ inputs.suggestions_per_comment }}
+        INPUT_REPO_PATH_PREFIX: ${{ inputs.repo_path_prefix }}
```
Comment (platisd) [inline action.yml:53]: Aren't these done implicitly?

Comment (renefritze) [reply action.yml:53]: Not with the composite action setup. 
See [this run](https://github.com/platisd/clang-tidy-pr-comments/actions/runs/7532032616/job/20501857199#step:7:119) before I noticed that (ref https://github.com/platisd/clang-tidy-pr-comments/pull/62/commits/c56fa2aa68c16d32316dcf021676541e9ac0ffe0)

**Comment by renefritze** [comment]
Noticed I still had a `!fixup` commit in here, squashed that and force pushed.

**Comment by platisd** [comment]
Sorry for the force pushing. Tested out the PR [here](https://github.com/platisd/clang-tidy-pr-comments/actions/runs/7532510587).

**Comment by platisd** [comment]
OK, I squashed some commits to make it more clear what is being done in this PR, hope that's fine with your @renefritze

**Comment by renefritze** [comment]
> OK, I squashed some commits to make it more clear what is being done in this PR, hope that's fine with your @renefritze

Sure thing.

**Review by platisd — APPROVED** [review]

**Comment by platisd** [comment]
Thanks for your efforts, it's now included in the latest release! :tada:

**Pull request merged**